python/paddle/distributed/checkpoint/save_state_dict.py

```diff
@@ -0,0 +1,186 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import List
+
+import paddle
+from paddle.distributed.communication.group import is_initialized
+from paddle.distributed.fleet.utils.log_util import logger
+
+from .metadata import LocalTensorIndex, LocalTensorMetadata, Metadata
+from .utils import compute_local_shape_and_global_offset, flatten_state_dict
+
+
+def check_state_dict(state_dict, process_group):
+    local_keys = list(state_dict.keys())
+    gloabl_keys = []
+    paddle.distributed.all_gather_object(gloabl_keys, local_keys, process_group)
+    for keys in gloabl_keys[1:]:
+        assert (
+            keys == gloabl_keys[0]
+        ), f"keys:{keys} != first_keys: {gloabl_keys[0]}"
+
+
+def check_file_name(file_name, process_group):
+    all_unique_id = []
+    unique_id = int(file_name.split(".")[0].split("_")[1])
+    paddle.distributed.all_gather_object(
+        all_unique_id, unique_id, process_group
+    )
+    for id in all_unique_id[1:]:
+        assert (
+            id == all_unique_id[0]
+        ), f"id:{id} !=  all_unique_id[0]:{file_name}"
+
+
+def merge_state_dict_metadata(global_state_dict):
+    assert isinstance(
+        global_state_dict, List
+    ), "The global_state_dict should be a list."
+    out = {}
+    for state_dict in global_state_dict:
+        for key, val in state_dict.items():
+            if key in out:
+                if val in out[key]:
+                    continue
+                out[key].append(val)
+            else:
+                out[key] = [val]
+    return out
+
+
+def dedup_storage_metadata(global_state_dict):
+    out = {}
+    for state_dict in global_state_dict:
+        for key, val in state_dict.items():
+            if key in out:
+                continue
+            out[key] = val
+    return out
+
+
+def save_state_dict(
+    state_dict,
+    path,
+    process_group=None,
+    coordinator_rank=0,
+) -> None:
+    """
+    Save the state_dict of model to path.
+
+    Args:
+        state_dict(Dict[str, paddle.Tensor]): The state_dict to save.
+        path(str): The directory to save state_dict.
+        process_group(paddle.distributed.collective.Group): ProcessGroup to be used for cross-rank synchronization. Use the default process group which contains all cards.
+        coordinator_rank(int): The rank used to save non distributed values. Rank0 is used by default.
+
+    Examples:
+        .. code-block:: python
```

Comment:
```suggestion
        .. code-block:: python
        
```
`code-block`下方得加空行，否则官网渲染会出错
![image](https://github.com/PaddlePaddle/Paddle/assets/70642955/7b02ad1f-1b66-41d5-bb32-294cc7737147)